libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -311,11 +311,11 @@ size_t PbftManager::dposEligibleVoteCount_(addr_t const &addr) {
   }
 }
 
-bool PbftManager::shouldSpeak(PbftVoteTypes type, uint64_t round, size_t step, size_t weighted_index) {
+bool PbftManager::shouldSpeak(PbftVoteTypes type, uint64_t round, size_t step) {
   // compute sortition
-  VrfPbftMsg msg(type, round, step, weighted_index);
+  VrfPbftMsg msg(type, round, step);
   VrfPbftSortition vrf_sortition(vrf_sk_, msg);
-  if (!vrf_sortition.canSpeak(sortition_threshold_, getDposTotalVotesCount())) {
+  if (!vrf_sortition.getBinominalDistribution(weighted_votes_count_, sortition_threshold_, getDposTotalVotesCount())) {
```

Comment:
In my understand, getBinominalDistribution is generate how many weight the vote will have. 
I think should be:
auto weight = vrf_sortition.getBinominalDistribution(weighted_votes_count_, sortition_threshold_, getDposTotalVotesCount());
VrfPbftMsg msg(type, round, step, weight);
VrfPbftSortition vrf_sortition(vrf_sk_, msg);
if (!vrf_sortition.canSpeak(sortition_threshold_, getDposTotalVotesCount())) {